File: LeetCode/Python Solutions/Unique Length-3 Palindromic Subsequences/unique.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        count = 0
        for char in set(s):  # Iterate over each unique character
            first = s.find(char)  # Find the first occurrence
            last = s.rfind(char)  # Find the last occurrence
            if first != last:  # Ensure there are characters in between
                count += len(set(s[first+1:last]))  # Count unique characters in between
        return count

File: LeetCode/Java Solutions/Unique Length-3 Palindromic Subsequences/unique.java
import java.util.HashSet;
import java.util.Set;

public class Solution {
    public int countPalindromicSubsequence(String s) {
        int count = 0;
        Set<Character> uniqueChars = new HashSet<>();

        // Add all unique characters to the set
        for (int i = 0; i < s.length(); i++) {
            uniqueChars.add(s.charAt(i));
        }

        // Iterate over each unique character
        for (char ch : uniqueChars) {
            int first = s.indexOf(ch);  // Find the first occurrence
            int last = s.lastIndexOf(ch);  // Find the last occurrence

            if (first != last) {  // Ensure there are characters in between
                Set<Character> middleChars = new HashSet<>();
                // Count unique characters in between
                for (int i = first + 1; i < last; i++) {
                    middleChars.add(s.charAt(i));
                }
                count += middleChars.size();
            }
        }

        return count;
    }
}
